python-api/tests/integrationtests/baseclass.py

```diff
@@ -0,0 +1,412 @@
+        Return when a match is found or timeout seconds have passed"""
+
+        filepath = os.path.join(
+            os.path.dirname(__file__),
+            'reference_frames',
+            filename)
+
+        self.log.debug("loading reference image %s", filepath)
+        expected = scipy.misc.imread(filepath).astype(float)
+
+        for turn in range(0, turns):
+            folder = tempfile.mkdtemp()
+
+            # build frame-fetching gstreamer pipeline
+            command = self.get_capture_command(
+                port, os.path.join(folder, "frame%d.png"))
+
+            self.log.info("running command for 0.5s (turn %u of %u)",
+                          turn, turns)
+            self.log.debug("%s", command)
+            proc = subprocess.Popen(shlex.split(command))
+            try:
+                proc.wait(timeout=timeout)
+                assert proc.poll() == 0
+            except subprocess.TimeoutExpired:
+                proc.kill()
+                proc.wait()
+
+            self.log.debug("command finished, comparing generated frames")
+            (success, img, diff) = self.compare_frames(expected, folder)
+
+            self.log.debug("removing temp-folder")
+            shutil.rmtree(folder)
+
+            if success:
+                break
+
+        if not success:
+            self.log.debug("last turn failed, "
+                           "saving images for human inspection")
+            self.save_images(filename, img, expected, diff)
+
+        assert success
+
+    def compare_frames(self, expected, folder):
+        """compare frames generated by the pipeline-command
+        against an expected image"""
+        for frame in sorted(glob(os.path.join(folder, "*.png"))):
+            # load generared image
+            img = scipy.misc.imread(frame).astype(float)
+
+            # Calulcate image-difference
+            diff = abs(img - expected)
+
+            # Calculate comparison-score
+            #   # Manhattan norm
+            #   m_norm = sum(abs(diff))
+            #
+            #   # Zero norm
+            #   z_norm = scipy.linalg.norm(diff.ravel(), 0) / img.size
+
+            # Maximal deviation between individual subpixels
+            max_dev = max(diff.ravel())
+
+            self.log.debug("comparison-results on frame %s: max_dev=%f",
+                           frame, max_dev)
+
+            # Compare calculated score
+            if max_dev <= 3:
+                # Score is small enough to succeed
+                self.log.info("comparison succeeded on frame %s "
+                              "with max_dev=%f", frame, max_dev)
+
+                return (True, img, diff)
+
+        self.log.info("comparison failed")
+        return (False, img, diff)
+
+    def expect_caps(self, port, expected):
+        """Fetch a sample-frame from the server and return its
+        caps as string"""
+        self.log.debug("building caps-expecting commandline")
+
+        command = ("gst-launch-1.0 "
```

Comment:
pylint / pep8 checkers are designed to keep us honest and help code be readable by keeping it consistent. They shouldn't be a straight jacket, where it makes sense to violate / deviate from them we shouldn't be afraid to do it. We should be making that choice consciously, rather than by accident however.